src/vector_of_array.jl

```diff
@@ -595,8 +604,14 @@ function Base.convert(::Type{Array}, VA::AbstractVectorOfArray)
 end
 
 # statistics
-@inline Base.sum(f, VA::AbstractVectorOfArray) = sum(f, Array(VA))
-@inline Base.sum(VA::AbstractVectorOfArray; kwargs...) = sum(Array(VA); kwargs...)
+@inline Base.sum(VA::AbstractVectorOfArray; kwargs...) = sum(identity, VA; kwargs...)
+@inline function Base.sum(f, VA::AbstractVectorOfArray; kwargs...)
+    if hasproperty(kwargs, :dims)
+        sum(Array(VA); kwargs...)
```

Comment:
This can be a potentially big allocation

Reply:
That was the default implementation, but I just realized two things:
- that method is probably not necessary
- mapreduce is implemented wrong

Reply:
Fixed in https://github.com/SciML/RecursiveArrayTools.jl/pull/325/commits/bdb58305372db57d38877e48157859e3137f7ccc